pkg/cmd/extension/manager.go

```diff
@@ -519,9 +534,24 @@ func (m *Manager) upgradeExtensions(exts []Extension, force bool) error {
 			fmt.Fprintf(m.io.Out, "%s\n", err)
 			continue
 		}
-		fmt.Fprintf(m.io.Out, "upgrade complete\n")
+		currentVersion := f.currentVersion
+		latestVersion := f.latestVersion
+		if !f.IsBinary() && len(currentVersion) > 8 {
+			currentVersion = currentVersion[:8]
+		}
+		if !f.IsBinary() && len(latestVersion) > 8 {
+			latestVersion = latestVersion[:8]
```

Comment:
Looks like the "truncate git SHA to 8 chars" logic is repeated 3 times throughout this package; maybe it's ripe for extracting into a small local helper? That way it would also be more obvious why is the version number getting truncated